src/plugins/midi-piano/midi-piano-display.js

```diff
@@ -0,0 +1,292 @@
+import PianoComponent from './piano-component.js';
+import { QuestionCircleOutlined } from '@ant-design/icons';
+import ClientConfig from '../../bootstrap/client-config.js';
+import { useService } from '../../components/container-context.js';
+import { sectionDisplayProps } from '../../ui/default-prop-types.js';
+import PlayIcon from '../../components/icons/media-player/play-icon.js';
+import PauseIcon from '../../components/icons/media-player/pause-icon.js';
+
+export default function MidiPianoDisplay({ content }) {
+
+  const keys = useRef([]);
+  const player = useRef(null);
+  const sampler = useRef(null);
+  const midiData = useRef(null);
+  const midiAccessObj = useRef(null);
+  const samplerHasLoaded = useRef(false);
+  const { NOTES } = midiPlayerNs.Constants;
+  const { sourceType, sourceUrl, midiTrackTitle, noteRange } = content;
+  const hasMidiFile = sourceUrl !== '';
+  const hasMidiTrackTitle = midiTrackTitle !== '';
+  const clientConfig = useService(ClientConfig);
+  const src = urlUtils.getMidiUrl({ cdnRootUrl: clientConfig.cdnRootUrl, sourceType, sourceUrl });
+
+  const getNoteNameFromMidiValue = midiValue => {
+    return NOTES[midiValue];
+  };
+
+  const getEventTypefromMidiCommand = (command, velocity) => {
+    switch (command) {
+      case 144:
+        if (velocity > 0) {
+          return 'Note on';
+        }
+        return 'Note off';
+      case 128:
+        return 'Note off';
+      case 176:
+        return 'Sustain off';
+      default:
+        return '';
+    }
+  };
+
+  function handleSamplerEvent(eventType, noteName) {
+    switch (eventType) {
+      case 'Note on':
+        sampler.current.triggerAttack(noteName);
+        break;
+      case 'Note off':
+        sampler.current.triggerRelease(noteName);
+        break;
+      default:
+        break;
+    }
+  }
+
+  function updateKeyVisualization(eventType, midiValue) {
+    if (!keys.current[midiValue]) {
+      return;
+    }
+    if (eventType === 'Note on') {
+      keys.current[midiValue].classList.add('ReactPiano__Key--active');
+      return;
+    }
+    if (eventType === 'Note off') {
+      keys.current[midiValue].classList.remove('ReactPiano__Key--active');
+    }
+  }
+
+  function handleMidiDeviceEvent(message) {
+    const noteName = getNoteNameFromMidiValue(message.data[1]);
+    const velocity = message.data.length > 2 ? message.data[2] : 0;
+    const command = message.data[0];
+    const eventType = getEventTypefromMidiCommand(command, velocity);
+    const midiValue = message.data[1];
+    updateKeyVisualization(eventType, midiValue);
+    handleSamplerEvent(eventType, noteName);
+  }
+
+  function onMIDISuccess(midiAccess) {
+    midiAccessObj.current = midiAccess;
+    for (const input of midiAccessObj.current.inputs.values()) {
+      input.onmidimessage = handleMidiDeviceEvent;
+    }
+  }
+
+  function onMIDIFailure() {
+    // eslint-disable-next-line no-console
+    console.log('Could not access your MIDI devices.');
```

Comment:
Let's use the `errorHandler.handleError` method here and not `console.log`.
Disabling an ESLint rule should be done as an exception, for the most part we want to respect the linter rules so that the code is nice and streamlined.

Reply:
This I tried to implement, but I don't know what to set `t` to. Please have a look at what I did, thank you.
Also, if I may ask just for me to learn: Why is the logger instantiated outside the function beneath the imports?